uiowa/uiowa
# [CCOM] Articles migration

# How to test



Sync a CCOM site in DevDesktop

```
ddev blt ds --site=default && ddev drush @default.local config-split:activate ccom_migrate -y && ddev drush @default.local uli /admin/config/sitenow/sitenow-migrate
```

Complete the form following the help text, including the Public Files Path field with the corresponding CCOM listing (such as `sites/medicine.uiowa.edu.pathology/files`)

`ddev drush @default.local ms`
Check that, among others, the `ccom_article` and `ccom_article_redirects` migrations have been registered.

`ddev drush @default.local mim ccom_article`
and, after
`ddev drush @default.local mim ccom_article_redirects`

Note: the `d7_files` migration can be run prior to these, but is not necessary.

I haven't run into a situation where it's needed for the size of migrations here, but you can use the `--limit=X` flag to migrate only `X` number of items at a time.

If a node being migrated had metatag info, you should see a terminal notice like:
<img width="889" alt="image" src="https://github.com/uiowa/uiowa/assets/49074566/ec26029c-2442-43ff-be57-d94e813ac9a0">
You can also check the `migrate_message_ccom_article` table for any logs, including metatag info not brought over.

The full field mappings can be found at https://github.com/uiowa/uiowa/issues/6640#issuecomment-1775833696. 

Spot check articles, and look for anything strange or out of place.
